flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/source/FlinkSource.java

```diff
@@ -247,6 +250,16 @@ int inferParallelism(FlinkInputFormat format, ScanContext context) {
       parallelism = Math.max(1, parallelism);
       return parallelism;
     }
+
+    private boolean localityEnabled() {
+      FileIO fileIO = table.io();
+      if (fileIO instanceof HadoopFileIO) {
+        Boolean localityConfig = readableConfig.get(FlinkConfigOptions
+            .TABLE_EXEC_ICEBERG_EXPOSE_SPLIT_LOCALITY_INFO);
+        return localityConfig != null ? localityConfig : true;
```

Comment:
Nit / open question:

On the subject of having a default value, it seems like the default for `HadoopFileIO` is `true`.

Should we also check that the path has a scheme that would make sense with locality enabled (e.g. the user isn't using HadoopFileIO with s3a or some other object store underneath)? Sort of like how Spark has the `FILESYSTEM_SUPPORT_LOCALITY` check?

I'm marking this as an open question, as it might be difficult to pass this information through, but I know some people who use Azure or even GCS who use the `HadoopFileIO` with the shim for their object storage (whatever is similar to `s3a` for their object storage), and they would not benefit from a default value of `true`.

Reply:
> I know some people who use Azure or even GCS who use the HadoopFileIO with the shim for their object storage

If this is the case, I think we should check the `scheme` like spark. @stevenzwu , could you please take a look at this?

Reply:
I thought the [discussion](https://lists.apache.org/thread/cknhjhvf63vp7xq7gwdbj9mlp0z0wj8b) is to use S3FileIO for Azure or GCS because they implements S3 like API.

With that said, I do agree to check the `scheme` like the Spark `Reader` class does. but @hililiwei we shouldn't need to open an InputFile for that purpose. We can just follow the Spark code.
```
 FileSystem fs = new Path(table.location()).getFileSystem(conf);
        fsscheme = fs.getScheme().toLowerCase(Locale.ENGLISH);
```